### DISCUSSION TITLE
Dynamic route parameters in browser URL get removed from nextjs router `asPath`

### DISCUSSION THREAD

**Discussion by therealgilles** [vercel/next.js#49476] [Help]
### Summary

Is it expected that dynamic route parameters present in the browser URL get automatically removed from nextjs router `asPath`? I understand they should never really be part of the browser URL but it seems also odd that they would get stripped off automatically.

I would think that the browser URL should be left as-is when reflected in `asPath`. This is what the documentation says:
```
asPath: String - The path as shown in the browser including the search params
and respecting the trailingSlash configuration. basePath and locale are not included.
```

It does not mention that any dynamic route parameters are stripped off.

**Comment by icyJoseph** [comment]
Could you give an example of what you see, and what you expected?

**Comment by jeff-r-skillrev** [comment]
I ran into this case, and noticed in the documentation that it recommends checking router.isReady before assuming that route parameters are available. When isReady is false, I see a "slug" name (such as "[projectId]"). When isReady is true, I see a dynamic value such as a BSON id "405bc66a05c82f75785ac55d" (I happen to be using Mongo on the backend) ).

source: https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object
```
Using the asPath field may lead to a mismatch between client and server if the page is rendered using server-side rendering or [automatic static optimization](https://nextjs.org/docs/pages/building-your-application/rendering/automatic-static-optimization). Avoid using asPath until the isReady field is true.
``` 

Once I coded for router.isReady === true, all worked well for me.

**Reply by therealgilles** [reply]
@jeff-r-koyaltech Thanks for responding. This is not what my issue is about.
